resources/assets/coffee/_classes/bar-chart.coffee

```diff
@@ -0,0 +1,96 @@
+###
+# Copyright 2016 ppy Pty. Ltd.
+#
+# This file is part of osu!web. osu!web is distributed with the hope of
+# attracting more community contributions to the core ecosystem of osu!.
+#
+# osu!web is free software: you can redistribute it and/or modify
+# it under the terms of the Affero GNU General Public License version 3
+# as published by the Free Software Foundation.
+#
+# osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+# warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+###
+
+class @BarChart
+  transition: 1000
+
+  margins:
+    top: 0
+    right: 0
+    bottom: 0
+    left: 0
+
+  constructor: (area, @options = {}) ->
+    @options.scales.x ||= d3.scale.linear()
```

Comment:
Looking again, this is wrong since it requires `scales` to be passed at all.

Something like this should work better (need check if actually works).

```coffeescript
options:
  scales:
    x: ...
    y: ...

constructor: (..., options = {}) ->
  _.merge @options, options
  ...
```